fix: remove normative references to ODRL and DCAT (#101)
eclipse-dataspace-protocol-base/DataspaceProtocol: specifications/negotiation/contract.negotiation.protocol.md

- The `callbackAddress` is a URL indicating where messages to the [=Consumer=] should be sent in asynchronous settings.
  If the address is not understood, the [=Provider=] MUST return an UNRECOVERABLE error.
- Different to a [=Catalog=] or [=Dataset=], the [=Offer=] inside

>   a [Contract Request Message](#contract-request-message) must have an `target` attribute. However, it's contained

Reviewer: ```suggestion
  a [Contract Request Message](#contract-request-message) must have a `target` attribute. However, it's contained
```

  appropriate `consumerPid`.
- The [=Dataset=] id is not required but can be included when the [=Provider=] initiates a CN.
- Different to a [=Dataset=],

>   the Offer inside a ContractOfferMessage must have an `target` attribute. However, it's contained Rules must not

Reviewer: ```suggestion
  the Offer inside a ContractOfferMessage must have a `target` attribute. However, its contained Rules must not
```
